[클린코드 1기 이희찬] 계산기 미션

amorfati0310 opened the pull request:
안녕하세요 클린코드 1기에 참여한 이희찬입니다.
[데모링크](https://amorfati0310.github.io/js-calculator/)
입력된 값을 누적하다가 `=` 연산자를 만날때 누적한 결과를 보여주도록 적용하였습니다.

🎯 기능 요구사항
- [x]  2개의 숫자에 대해 덧셈이 가능하다.
- [x]  2개의 숫자에 대해 뺄셈이 가능하다.
- [x]  2개의 숫자에 대해 곱셈이 가능하다.
- [x]  2개의 숫자에 대해 나눗셈이 가능하다.
- [x]  AC(All Clear)버튼을 누르면 0으로 초기화 한다.
- [x]  숫자는 한번에 최대 3자리 수까지 입력 가능하다.
- [x]  계산 결과를 표현할 때 소수점 이하는 버림한다.

amorfati0310 committed:
린트 설정 추가

amorfati0310 committed:
feat: 기능 구현

amorfati0310 committed:
feat: TestCase 추가

amorfati0310 committed:
refactor: 중첩 if 제거

amorfati0310 committed:
feat: 0 일 때 처리 적용

amorfati0310 committed:
fix: 0 일때 처리 잘 못 적용한 부분 수정

glassyi commented on src/js/components/Digits.js:
```diff
@@ -0,0 +1,33 @@
+  }
+
+  handleClick({ target, onClick }) {
+    if (target.className !== 'digit') {
```
element에 이벤트 바인딩하는 코드를 각각 class로 만드신 이유가 궁금하네요~ 🙂
Digits, Operations, Modifiers 클래스의 내용이 
target.className 에 대한 비교 빼고 동일해서요~
하나의 클래스나 함수로 만들 수 있지 않을까요?

그리고 제가 생각했을 때
target class를 인스턴스 생성할 때 받고, init에서 해당 element가 있는지 체크하고 있으니까 
여기서 따로 또 비교하지 않아도 될 것 같아요! 
그럼 target을 넘길 필요가 없고 onClick 함수 실행도 
bindEvents 메소드의 이벤트 핸들러 안에서 하면 되지 않을까 생각합니다!

혹시 다른 의도가 있으셨다면 코멘트 남겨주세요!!

amorfati0310 replied:
@glassyi  안녕하세요 꼼꼼한 리뷰 감사합니다. 
1. 기존의 마크업이 있어서 리스트 클릭 이벤트만 처리를 하다 보니 동일 함수로 처리해도 될 것 같긴 하네요.  
이 부분 `viewComponent`가 UI 적으로 분리되는게 구분이 더 잘 될 것 같아서 분리시켰습니다 :D 

2. `element`로 바로 내려주는게 더 낫겠네요 👍  

3. 아 이 부분은 `delegation`이 걸려 있어서 컨테이너를 클릭하고(`digits`) 해당 `digit`같은 영역을 클릭하지 않았을때 같은 경우에는 유효한 코드가 될 것 같아요 
그런데 괜히 `handleClick`으로  감싸준 것 같네요 바로 처리하는게 더 깔끔해보이네요

glassyi commented on src/js/components/Total.js:
```diff
@@ -0,0 +1,12 @@
+  }
+
+  render({ stack }) {
+    this.element.innerText = stack.length ? stack.join(' ') : 0;
```
stack.join(' ') 에 space가 들어가서 올려주신 계산기 데모로 보면 
예를 들어 30/3= 10의 결과가 나온 상태에서 다시 숫자를 누르면(5를 눌렀다고 가정)
10 5 <- 요렇게 사이에 space가 들어갑니다~ 
수정되면 좋을 것 같습니다 🙂
```suggestion
    this.element.innerText = stack.length ? stack.join('') : 0;
```

amorfati0310 replied:
그러네요 감사합니다 수정해놓을게요

glassyi commented:
안녕하세요~ 이희찬님 리뷰를 맡게된 임유리입니다!

희찬님 코드 보면서 궁금한 부분들과 수정되면 좋을 것 같은 부분 코멘트 남겼습니다! 감사합니다!

amorfati0310 committed:
feat: 리뷰 반영

- 컴포넌트 클래스 메소드 가독성 및 괜히 감싸져 있는 부분 제거
- element로 바로 전달받도록 수정
- 결과 값 바로 숫자 붙였을때 이전 값 제거 되는 이슈 수정

amorfati0310 commented:
> 안녕하세요~ 이희찬님 리뷰를 맡게된 임유리입니다!
> 
> 희찬님 코드 보면서 궁금한 부분들과 수정되면 좋을 것 같은 부분 코멘트 남겼습니다! 감사합니다!

@glassyi 님 감사합니다 리뷰 반영하여 수정하였습니다.

glassyi commented:
바로 머지하겠습니다~ 고생하셨습니다!
